andela/neon-ah-backend
# #163590145 Enable case insensitivity

#### What does this PR do?
Enable case insensitivity across resources

#### Description of Task to be completed?
Enable users to query db with case-insensitive parameters and achieve desired results:
 •  Convert username & email search or entry into db to lowercase
 •  Convert tag name search or entry into db to lowercase

#### How should this be manually tested?
 •  Clone this branch
 •  CD into the repo
 •  Run `npm install`
 •  Run `npm run migrate && npm run seed`
 •  Run `npm run dev`
 •  Use postman to access the following endpoints:
```
 •  **http://localhost/3000/auth/signup (POST)**
    •  Signup with case insensitive username and/or email
 
 •  **http://localhost/3000/auth/verify/:token (POST)**
    •  Copy the token from your email and add to the above route

 •  **http://localhost/3000/auth/login (POST)**
    •  Login with the above endpoint using sensitive or case-insensitive username or email
 
 •  **http://localhost/3000/articles (POST)***
    • Create an article with the above endpoint with the parameters below:
        • title, content, tagsArray(case-insensitive comma separated strings)
    • You'll see the tags in the response appear in all-lowercase

```
#### What are the relevant pivotal tracker stories?
[Enable case insensitivity](https://www.pivotaltracker.com/story/show/163590145)
